plugins/modules/job_invocation.py

```diff
@@ -0,0 +1,222 @@
+    description:
+      - Control concurrency level and distribution over time
+    type: dict
+    suboptions:
+      time_span:
+        description:
+          - Distribute tasks over given number of seconds
+        type: int
+      concurrency_level:
+        description:
+          - Maximum jobs to be executed at once
+        type: int
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.entity_state
+  - theforeman.foreman.foreman.taxonomy
+'''
+
+EXAMPLES = '''
+
+- name: "Run remote command on a single host once"
+  job_invocation:
+    search_query: "name ^ (foreman.example.com)"
+    command: 'ls'
+    job_template: "Run Command - SSH Default"
+    ssh:
+      effective_user: "tester"
+
+- name: "Run ansible commad on active hosts once a day"
+  job_invocation:
+    bookmark: 'active'
+    command: 'pwd'
+    job_template: "Run Command - Ansible Default"
+    recurrence:
+      cron_line: "30 2 * * *"
+    concurrency_control:
+      concurrency_level: 2
+'''
+
+RETURN = ''' # '''
+
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import (
+    ForemanTaxonomicEntityAnsibleModule,
+)
+
+ssh_foreman_spec = {
+    'effective_user': dict(),
+}
+
+recurrence_foreman_spec = {
+    'cron_line': dict(),
+    'max_iteration': dict(type='int'),
+    'end_time': dict(),
+}
+
+scheduling_foreman_spec = {
+    'start_at': dict(),
+    'start_before': dict(),
+}
+
+concurrency_control_foreman_spec = {
+    'time_span': dict(type='int'),
+    'concurrency_level': dict(type='int'),
+}
+
+
+class ForemanJobInvocationModule(ForemanTaxonomicEntityAnsibleModule):
+    pass
+
+
+def main():
+    module = ForemanJobInvocationModule(
+        foreman_spec=dict(
+            search_query=dict(),
+            bookmark=dict(type='entity'),
+            job_template=dict(required=True, type='entity'),
+            targeting_type=dict(default='static_query', choices=['static_query', 'dynamic_query']),
+            randomized_ordering=dict(type='bool'),
+            feature=dict(),
+            command=dict(),
+            inputs=dict(type='dict'),
+            execution_timeout_interval=dict(type='int'),
+            ssh=dict(type='dict', options=ssh_foreman_spec),
+            recurrence=dict(type='dict', options=recurrence_foreman_spec),
+            scheduling=dict(type='dict', options=scheduling_foreman_spec),
+            concurrency_control=dict(type='dict', options=concurrency_control_foreman_spec),
+        ),
+        required_one_of=[['search_query', 'bookmark']],
+        required_if=[
+            ['job_template', 'Run Command - SSH Default', ['command']],
+            ['job_template', 'Run Command - Ansible Default', ['command']],
+        ],
+        entity_opts={'search_by': 'description'}
+    )
+
+    # command input required by api
+    if 'command' in module.foreman_params:
+        module.foreman_params['inputs'] = {"command": module.foreman_params.pop('command')}
```

Comment:
Would it be a big burden on the user if we required, that a command must be specified in the `inputs` section?
It would take away some complexity of this module.
To get even more complexity back, we could add some generic input validation here. (<- Thinking out loud)

Reply:
I don't know, I tried to mimic the UI experience as much as possible. I guess it's ok to have it under inputs, if that's the majority opinion :) Not sure how generic we can get with the input validation as every additional template would require different inputs

Reply:
My idea was to fetch the plugin in question from the server to get that info. But we should consider that a future improvement.

Reply:
@evgeni opinion on this?

Reply:
I can see the beauty of mimicking the UI, but also of keeping it simple.

that said, I don't think these additional two lines are adding much complexity?

Reply:
okay, 5 lines, I forgot the `required_if` above…